swagger_templates/python/api_client.mustache

```diff
@@ -0,0 +1,620 @@
+    Do not edit the class manually.
+
+    :param host: The base path for the server to call.
+    :param header_name: a header to pass when making calls to the API.
+    :param header_value: a header value to pass when making calls to the API.
+    """
+    def __init__(self, host=None, header_name=None, header_value=None, cookie=None):
+
+        """
+        Constructor of the class.
+        """
+        self.rest_client = RESTClientObject()
+        self.default_headers = {}
+        if header_name is not None:
+            self.default_headers[header_name] = header_value
+        if host is None:
+            self.host = Configuration().host
+        else:
+            self.host = host
+        self.cookie = cookie
+        # Set default User-Agent.
+        self.user_agent = '{{#httpUserAgent}}{{{.}}}{{/httpUserAgent}}{{^httpUserAgent}}Swagger-Codegen/{{{packageVersion}}}/python{{/httpUserAgent}}'
+
+    @property
+    def user_agent(self):
+        """
+        Gets user agent.
+        """
+        return self.default_headers['User-Agent']
+
+    @user_agent.setter
+    def user_agent(self, value):
+        """
+        Sets user agent.
+        """
+        self.default_headers['User-Agent'] = value
+
+    def set_default_header(self, header_name, header_value):
+        self.default_headers[header_name] = header_value
+
+    def __call_api(self, resource_path, method,
+                   path_params=None, query_params=None, header_params=None,
+                   body=None, post_params=None, files=None,
+                   response_type=None, auth_settings=None, callback=None,
+                   _return_http_data_only=None, collection_formats=None):
+
+        # header parameters
+        header_params = header_params or {}
+        header_params.update(self.default_headers)
+        if self.cookie:
+            header_params['Cookie'] = self.cookie
+        if header_params:
+            header_params = self.sanitize_for_serialization(header_params)
+            header_params = dict(self.parameters_to_tuples(header_params,
+                                                           collection_formats))
+
+        # path parameters
+        if path_params:
+            path_params = self.sanitize_for_serialization(path_params)
+            path_params = self.parameters_to_tuples(path_params,
+                                                    collection_formats)
+            for k, v in path_params:
+                v_str = str(v)
+                # Check for the special case of the
+                # x-isi-url-encode-path-param, which indicates that the
+                # parameter should be encoded with quote_plus in order
+                # to encode the '/' character.
+                quote_plus_tag = "__x-isi-url-encode-path-param__"
+                tag_len = len(quote_plus_tag)
+                # check if the first part of v_str matches the tag
+                if v_str[:tag_len] == quote_plus_tag:
```

Comment:
Here's how where I changed how it deals with the x-isi-url-encoded-path-param tag.